apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -639,26 +638,23 @@ const TeacherApplication = React.createClass({
       }
     });
 
-    if (validateDistrictData(districtValues)) {
-      document.getElementById('district-error-placeholder').innerHTML = '';
+    if (validateDistrictData(this.props.schoolDistrictData)) {
+      this.props.districtErrorMessageHandler('');
     } else {
       topInvalidElementId = 'district-error-placeholder';
-      ReactDOM.render(
-        <label style={{color: 'red', fontSize: '14pt', fontWeight: 'bold'}}>
-          Please complete this section with your school's information
-        </label>,
-        document.getElementById(topInvalidElementId)
-      );
+      this.props.districtErrorMessageHandler("Please complete this section with your school's information");
     }
 
     if (topInvalidElementId) {
       this.setState({submitting: false});
       let topInvalidElement = document.getElementById(topInvalidElementId);
```

Comment:
if this element is getting rendered by react, I would use `refs` to get the dom node rather than getElementById. But that's probably a change for a future diff.